### PR TITLE
MMF-1778

### PR DESCRIPTION
**Pull request by guillaume-dequenne-sonarsource**

### CODE REVIEW

File: sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    firstPrimaryLoc = first.primaryLocation();
+    assertThat(firstPrimaryLoc.inputComponent().key()).isEqualTo(PYLINT_FILE);
+    assertThat(firstPrimaryLoc.message())
+      .isEqualTo("Missing docstring");
+  }
+
+  @Test
+  public void no_issues_unknown_files() throws IOException {
+    List<ExternalIssue> externalIssues = executeSensorImporting(7, 9, "pylint_report_unknown_files.txt");
+    assertThat(externalIssues).isEmpty();
```
Comment (pynicolas) [inline sonar-python-plugin/src/test/java/org/sonar/plugins/python/pylint/PylintSensorTest.java:198]: Should we assert anything about the logs?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/ExternalIssuesSensor.java [hunk 2]
```diff
@@ -33,28 +33,41 @@
...
-      .onlyWhenConfiguration(conf -> conf.hasKey(reportPathKey()))
+      .onlyWhenConfiguration(conf -> conf.hasKey(reportPathKey()) || hasPylintLegacyConfig(conf))
       .onlyOnLanguage(Python.KEY)
       .name("Import of " + linterName() + " issues");
   }
 
+  private boolean hasPylintLegacyConfig(Configuration conf) {
+    // Only for PylintSensor
+    return reportPathKey().equals(PylintSensor.REPORT_PATH_KEY) &&
+      conf.hasKey(PYLINT_LEGACY_KEY);
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/ExternalIssuesSensor.java:60]: Alternative solution: rely on polymorphism and define this condition in PylintSensor.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/bandit/BanditSensor.java [hunk 3]
```diff
@@ -51,15 +51,11 @@
...
-      boolean engineIdIsSupported = context.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(7, 4));
-      BanditJsonReportReader.read(in, issue -> saveIssue(context, issue, unresolvedInputFiles, engineIdIsSupported));
-    } catch (IOException | ParseException | RuntimeException e) {
-      LOG.error("No issues information will be saved as the report file '{}' can't be read. " +
-        e.getClass().getSimpleName() + ": " + e.getMessage(), reportPath, e);
-    }
+  protected void importReport(File reportPath, SensorContext context, Set<String> unresolvedInputFiles) throws IOException, ParseException {
+    InputStream in = new FileInputStream(reportPath);
+    LOG.info("Importing {}", reportPath);
+    boolean engineIdIsSupported = context.getSonarQubeVersion().isGreaterThanOrEqual(Version.create(7, 4));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/bandit/BanditSensor.java:57]: Indeed, I had in mind to have a small clean-up PR afterwards to freshen a bit `BanditSensor` among other things, didn't want to mix that with that with the actual implementation for ease of reviewing.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/TextReportReader.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    String ruleKey = m.group(4);
+    String message = m.group(5);
+    return new Issue(filePath, ruleKey, message, lineNumber, columnNumber);
+  }
+
+  private static Issue extractLegacyStyleIssue(Matcher m) {
+    String filePath = m.group(1);
+    int lineNumber = Integer.parseInt(m.group(2));
+    String ruleKey = m.group(3);
+    int lastIndex = Math.min(ruleKey.indexOf(","), ruleKey.indexOf("("));
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/TextReportReader.java:94]: I thought I saw such format somewhere, but impossible to find it again...I'll remove this.

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java [hunk 5]
```diff
@@ -1,137 +0,0 @@
...
-    if (!sensorContext.activeRules().findByRepository("Pylint").isEmpty()) {
-      analysisWarnings.addUnique("Deprecation notice and future breaking changes: The import of Pylint issues will soon change. " +
-        "Please follow the instructions in documentation’s section \"Analyzing Source Code\" >> \"Languages\" >> \"Python\" >> \"Pylint\".");
-    }
-    if (!shouldExecute() || !prepareWorkDir(workDir) || !initializeAnalyzer(sensorContext)) {
-      return;
-    }
-
-    LOG.warn("Execution of Pylint is deprecated and will be removed." +
-      " Instead, Pylint should be executed before sonar-scanner and its report should be imported using the '" + PylintImportSensor.REPORT_PATH_KEY + "' property.");
```
[the remark this answers is not in the document]
Comment (guillaume-dequenne-sonarsource) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/pylint/PylintSensor.java:80]: After discussion with Alex, we agreed that it's not needed as we have displayed a deprecation warning for quite some time now.

**Pull request merged**